lib/git_hooks/pre_commit/rubocop.rb

```diff
@@ -1,32 +1,73 @@
 module GitHooks
   module PreCommit
     class Rubocop
-      attr_reader :git_repository, :rubocop_validator
+      RUBOCOP_STASH_NAME = 'rubocop-stash'
 
-      def self.validate
+      def self.validate(options = {})
         new(
-          GitHooks.configurations.git_repository, RubocopValidator.new
+          GitHooks.configurations.git_repository,
+          RubocopValidator.new,
+          options
         ).validate
       end
 
-      def initialize(git_repository, rubocop_validator)
-        @git_repository, @rubocop_validator = git_repository, rubocop_validator
+      def initialize(git, rubocop_validator, options = {})
+        @git = git
+        @rubocop_validator = rubocop_validator
+        @options = options
       end
 
       def validate
         abort 'Check rubocop offences' if offences?
       end
 
+      protected
+
+      attr_reader :git, :rubocop_validator, :options
+
       private
 
       def changed_files
-        git_repository
+        git
           .added_or_modified
-          .select { |file| File.extname(file) == '.rb' }
+          .select(&ruby_files)
       end
 
       def offences?
-        rubocop_validator.errors?(changed_files)
+        stash_me_maybe { rubocop_validator.errors?(changed_files) }
+      end
+
+      def ruby_files
+        -> (file) { File.extname(file) == '.rb' }
+      end
+
+      def stash_me_maybe(&blk)
```

Comment:
I'm using this style **a lot** recently. Methods returning lambdas and methodos wrapping blocks are just so damn elegant in Ruby.